src/node_sqlite.cc

```diff
@@ -1000,6 +1000,14 @@ void DatabaseSync::Close(const FunctionCallbackInfo<Value>& args) {
   db->connection_ = nullptr;
 }
 
+void DatabaseSync::Dispose(const v8::FunctionCallbackInfo<v8::Value>& args) {
+  v8::TryCatch try_catch(args.GetIsolate());
+  Close(args);
+  if (try_catch.HasCaught()) {
+    CHECK(try_catch.CanContinue());
+  }
```

Comment:
Swallowing the error here is likely not ideal but it's what the current code does so I'm preserving that here. Ideally the error would be propagated instead.

Reply:
I would assume that the swallow here is intentional, so users can do early `session.close()` within the scope, and exiting the scope won't throw. We might still prefer to re-throw anything except `ERR_INVALID_STATE` though.
Either way, I think it would be worth adding a TODO comment here.